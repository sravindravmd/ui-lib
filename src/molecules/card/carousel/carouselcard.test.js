import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import CarouselCard from './index'

afterEach(cleanup)

describe('<CarouselCard/>', () => {
  const imgUrl =
    'https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60'
  it('renders CarouselCard correctly', () => {
    const { container } = render(
      <CarouselCard
        title='V-neck Ruffle Top in Floral Sunset'
        subtitle='921403392'
        imgUrl={imgUrl}
        imgAltText='card-img'
        onClick={() => {}}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render CarouselCard after click', () => {
    const { container, getByText } = render(
      <CarouselCard
        title='V-neck Ruffle Top in Floral Sunset'
        subtitle='921403392'
        imgUrl={imgUrl}
        imgAltText='card-img'
        onClick={() => {}}
      />
    )
    fireEvent.click(getByText('V-neck Ruffle Top in Floral Sunset'))
    expect(container).toMatchSnapshot()
  })
})
