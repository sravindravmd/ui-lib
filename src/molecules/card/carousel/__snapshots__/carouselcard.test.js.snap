// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselCard/> renders CarouselCard correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ hBvqpI"
  >
    <div
      class="sc-jSgupP cZTZDh"
      tabindex="-1"
    >
      <img
        alt="card-img"
        class="sc-gsTCUz eRPChC"
        src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
      />
      <div
        class="sc-dlfnbm hFkLbG"
      >
        <div
          class="sc-hKgILt bDrTGL"
        >
          V-neck Ruffle Top in Floral Sunset
        </div>
        <div
          class="sc-eCssSg gRwfES"
        >
          921403392
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CarouselCard/> should render CarouselCard after click 1`] = `
<div>
  <div
    class="sc-bdfBwQ gOzfJa"
  >
    <div
      class="sc-jSgupP cZTZDh"
      tabindex="-1"
    >
      <img
        alt="card-img"
        class="sc-gsTCUz eRPChC"
        src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
      />
      <div
        class="sc-dlfnbm hFkLbG"
      >
        <div
          class="sc-hKgILt bDrTGL"
        >
          V-neck Ruffle Top in Floral Sunset
        </div>
        <div
          class="sc-eCssSg gRwfES"
        >
          921403392
        </div>
      </div>
    </div>
  </div>
</div>
`;
